src/rospack.cpp

```diff
@@ -1647,15 +1661,16 @@ Rosstackage::computeDepsInternal(Stackage* stackage, bool ignore_errors, const s
       {
         continue;
       }
+      result = false;
       if(ignore_errors)
       {
         Stackage* dep =  new Stackage(dep_pkgname, "", "", "");
         stackage->deps_.push_back(dep);
       }
-      else
+      else if (!quiet_)
       {
         std::string errmsg = get_manifest_type() + " '" + stackage->name_ + "' depends on non-existent package '" + dep_pkgname + "' and rosdep claims that it is not a system dependency. Check the ROS_PACKAGE_PATH or try calling 'rosdep update'";
-        throw Exception(errmsg);
```

Comment:
Existing code still aims to catch this exception. Should all calling code which doesn't throw an exception anymore be updated? Mostly to make this this doesn't result in a undesired change of behavior?

Reply:
Unless I've missed something I have updated all methods in this package that call computeDeps to use the return value instead (I've used CLions `Find usages` to make sure I get all usages).   
I've just checked again and unless you're calling `computeDeps`, `computeDepsInternal` or `depsWhyDetail` directly, you wouldn't get the exception anyway since all other methods catch them. These methods are all private so there shouldn't be anyone calling them outside of the package.

Generally, this, of course, is a change of behavior but I would say it is a desired change.
The public methods behave as before except that they now produce results rather than returning without producing results.
The only exception is `depsOnDetail` which ignores the result of `computeDeps` since, if I'm not mistaken, it scans the entire stackage and would always return false if any package in the stackage is broken. I don't know anymore why I thought returning true in that case would be better, though. I can change it back to returning false if a package is broken if you'd prefer that?

Reply:
I was aiming for existing code which has a `try` / `catch` logic. Before this patch some functions where raising exception. The question is are there now any cases where those `try` / `catch` logic is obsolete because nothing in the try block can throw anymore?

Reply:
Yes, at a glance there are some that are most likely obsolete now. I can remove them but as a rather defensive programmer, I would lean towards keeping them just in case; unless there's a reason to remove them that I'm missing.